raiden/tests/integration/fixtures/blockchain.py

```diff
@@ -46,65 +48,67 @@ def web3(
     keys_to_fund.add(deploy_key)
     keys_to_fund = sorted(keys_to_fund)
 
-    if blockchain_type == 'geth':
-        host = '0.0.0.0'
-        rpc_port = blockchain_rpc_ports[0]
-        endpoint = f'http://{host}:{rpc_port}'
-        web3 = Web3(HTTPProvider(endpoint))
-
-        assert len(blockchain_private_keys) == len(blockchain_rpc_ports)
-        assert len(blockchain_private_keys) == len(blockchain_p2p_ports)
-
-        geth_nodes = [
-            GethNodeDescription(
-                key,
-                rpc,
-                p2p,
-                miner=(pos == 0),
-            )
-            for pos, (key, rpc, p2p) in enumerate(zip(
-                blockchain_private_keys,
-                blockchain_rpc_ports,
-                blockchain_p2p_ports,
-            ))
-        ]
-
-        accounts_to_fund = [
-            privatekey_to_address(key)
-            for key in keys_to_fund
-        ]
-
-        base_datadir = str(tmpdir)
-
-        if _GETH_LOGDIR:
-            base_logdir = os.path.join(_GETH_LOGDIR, request.node.name)
-        else:
-            base_logdir = os.path.join(base_datadir, 'logs')
-
-        geth_processes = geth_run_private_blockchain(
-            web3=web3,
-            accounts_to_fund=accounts_to_fund,
-            geth_nodes=geth_nodes,
-            base_datadir=base_datadir,
-            log_dir=base_logdir,
-            chain_id=chain_id,
-            verbosity=request.config.option.verbose,
-            random_marker=random_marker,
-        )
+    if blockchain_type not in ('geth', 'parity'):
+        raise ValueError(f'unknown blockchain_type {blockchain_type}')
 
-        yield web3
+    host = '0.0.0.0'
+    rpc_port = blockchain_rpc_ports[0]
+    endpoint = f'http://{host}:{rpc_port}'
+    web3 = Web3(HTTPProvider(endpoint))
+
+    assert len(blockchain_private_keys) == len(blockchain_rpc_ports)
+    assert len(blockchain_private_keys) == len(blockchain_p2p_ports)
+
+    eth_nodes = [
+        EthNodeDescription(
+            key,
```

Comment:
please use kwargs here. I know you just moved the code but it's a good chance to add them :)